server/apps/newsletter/services.py

```
@@ -0,0 +1,74 @@
+        endpoint = f"{self.base_url}/publications/{self.publication_id}/subscriptions/email:{email}"
+        
+        data = {
+            "status": "active" if is_active else "inactive"
+        }
+        
+        try:
+            response = requests.patch(endpoint, headers=self.headers, json=data)
+            response.raise_for_status()
+            return response.json()
+        except requests.exceptions.RequestException as e:
+            raise Exception(f"Failed to update Beehiiv subscriber: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)